enhanced/metrics.go

```diff
@@ -0,0 +1,756 @@
+			[]string{"instance", "region"},
+			map[string]string(nil),
+		),
+	},
+	"rdsosmetrics_memory_hugePagesRsvd": {
+		Name: "hugePagesRsvd",
+		Desc: prometheus.NewDesc(
+			"rdsosmetrics_memory_hugePagesRsvd",
+			"The number of committed huge pages.",
+			[]string{"instance", "region"},
+			map[string]string(nil),
+		),
+	},
+	"rdsosmetrics_memory_mapped": {
+		Name: "mapped",
+		Desc: prometheus.NewDesc(
+			"rdsosmetrics_memory_mapped",
+			"The total amount of file-system contents that is memory mapped inside a process address space, in kilobytes.",
+			[]string{"instance", "region"},
+			map[string]string(nil),
+		),
+	},
+	"rdsosmetrics_memory_total": {
+		Name: "total",
+		Desc: prometheus.NewDesc(
+			"rdsosmetrics_memory_total",
+			"The total amount of memory, in kilobytes.",
+			[]string{"instance", "region"},
+			map[string]string(nil),
+		),
+	},
+	"rdsosmetrics_memory_writeback": {
+		Name: "writeback",
+		Desc: prometheus.NewDesc(
+			"rdsosmetrics_memory_writeback",
+			"The amount of dirty pages in RAM that are still being written to the backing storage, in kilobytes.",
+			[]string{"instance", "region"},
+			map[string]string(nil),
+		),
+	},
+	"rdsosmetrics_memory_buffers": {
+		Name: "buffers",
+		Desc: prometheus.NewDesc(
+			"rdsosmetrics_memory_buffers",
+			"The amount of memory used for buffering I/O requests prior to writing to the storage device, in kilobytes.",
+			[]string{"instance", "region"},
+			map[string]string(nil),
+		),
+	},
+	"rdsosmetrics_memory_dirty": {
+		Name: "dirty",
+		Desc: prometheus.NewDesc(
+			"rdsosmetrics_memory_dirty",
+			"The amount of memory pages in RAM that have been modified but not written to their related data block in storage, in kilobytes.",
+			[]string{"instance", "region"},
+			map[string]string(nil),
+		),
+	},
+	"rdsosmetrics_memory_free": {
+		Name: "free",
+		Desc: prometheus.NewDesc(
+			"rdsosmetrics_memory_free",
+			"The amount of unassigned memory, in kilobytes.",
+			[]string{"instance", "region"},
+			map[string]string(nil),
+		),
+	},
+	"rdsosmetrics_memory_active": {
+		Name: "active",
+		Desc: prometheus.NewDesc(
+			"rdsosmetrics_memory_active",
+			"The amount of assigned memory, in kilobytes.",
+			[]string{"instance", "region"},
+			map[string]string(nil),
+		),
+	},
+	"rdsosmetrics_memory_hugePagesSurp": {
+		Name: "hugePagesSurp",
+		Desc: prometheus.NewDesc(
+			"rdsosmetrics_memory_hugePagesSurp",
+			"The number of available surplus huge pages over the total.",
+			[]string{"instance", "region"},
+			map[string]string(nil),
+		),
+	},
+	"rdsosmetrics_processList_vss": {
+		Name: "vss",
+		Desc: prometheus.NewDesc(
+			"rdsosmetrics_processList_vss",
+			"The amount of virtual memory allocated to the process, in kilobytes.",
+			[]string{"instance", "region", "id"},
+			map[string]string(nil),
+		),
+	},
+	"rdsosmetrics_processList_vmlimit": {
+		Name: "vmlimit",
+		Desc: prometheus.NewDesc(
+			"rdsosmetrics_processList_vmlimit",
+			"TODO",
```

Comment:
@AlekSi some metrics are missing in http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_Monitoring.OS.html